### DISCUSSION TITLE
Xstate React throws errors and has circular dependencies when bundled with Rollup.

### DISCUSSION THREAD

**Discussion by ghattrell** [statelyai/xstate#2156] [Ideas]
The @xstate/react package still seems to have some problems when bundled with rollup. I believe they have been fixed in the main xstate package. (https://github.com/davidkpiano/xstate/issues/498), but not in this package.

Are there any plans to tweak the build configuration for this package in line with xstate build configs so that it can be used out the box with rollup?

For anyone experiencing this problem the fix is fairly simple just target the lib build rather than the es build. However, it would be lovely to be able to use xstate react with the module build.

**Comment by Andarist** [comment]
The reported issue is really vague and doesn't mention a lot of specifics about the problem. Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox.

**Reply by ghattrell** [reply]
Apologies, it is literally the exact same build errors as was opened on the xstate repo here (https://github.com/davidkpiano/xstate/issues/498) so thought that was overkill. I will get a basic repo configured in the coming days.

**Reply by ghattrell** [reply]
```javascript
(!) `this` has been rewritten to `undefined`
https://rollupjs.org/guide/en/#error-this-is-undefined
node_modules/@xstate/react/es/useService.js
1: var __read = (this && this.__read) || function (o, n) {
                 ^
2:     var m = typeof Symbol === "function" && o[Symbol.iterator];
3:     if (!m) return o;
...and 1 other occurrence
node_modules/@xstate/react/es/useInterpret.js
1: var __assign = (this && this.__assign) || function () {
                   ^
2:     __assign = Object.assign || function(t) {
3:         for (var s, i = 1, n = arguments.length; i < n; i++) {
...and 5 other occurrences
node_modules/@xstate/react/es/useMachine.js
1: var __read = (this && this.__read) || function (o, n) {
                 ^
2:     var m = typeof Symbol === "function" && o[Symbol.iterator];
3:     if (!m) return o;
...and 3 other occurrences
...and 4 other files
(!) Circular dependency
node_modules/@xstate/react/es/useMachine.js -> node_modules/@xstate/react/es/useInterpret.js -> node_modules/@xstate/react/es/useReactEffectActions.js -> node_modules/@xstate/react/es/useMachine.js
```
